server/app/controllers/applicant/ApplicantInformationController.java

```diff
@@ -90,7 +94,8 @@ public CompletionStage<Result> setLangFromBrowser(Http.Request request, long app
                             /* page= */ Optional.of(1))
                         .url();
               } else {
-                redirectLink = routes.ApplicantProgramsController.index(applicantId).url();
+                CiviFormProfile profile = profileUtils.currentUserProfile(request).orElseThrow();
```

Comment:
**suggestion (blocking):** This PR uses `orElseThrow` quite a bit. It'd be helpful for each of those places to include an error message that will assist in error reporting and debugging when a civic entity reports a incident. Even something as simple as `orElseThrow(() -> new RuntimeException("No profile found!")`

Reply:
Added a `MissingOptionalException` and used it throughout.